Code-4-Community/speak-for-the-trees-frontend
# Frontend Capabilities to Delete a Stewardship Activity

## Why

[ClickUp Ticket](https://app.clickup.com/t/1dk44y0)



This change allows users to delete their created stewardship activities. It creates a user interface for existing backend capabilities and adds flexibility for incorrectly inputted/extraneous stewardship activities, thus allowing for increased website accuracy.

## This PR

This change involved the creation and styling of a delete icon that is visible only for a given user's own trees and admin. Once users click this button, a modal pops up to confirm their deletion. From there, the stewardship activity is deleted and no longer visible in the list of documented stewardship activities.

## Screenshots

Activity Display Window
<img width="386" alt="Screen Shot 2023-01-21 at 6 48 56 PM" src="https://user-images.githubusercontent.com/30560773/213894518-2872bbd7-2800-481c-994d-fc7a295031c0.png">

On hover
<img width="387" alt="Screen Shot 2023-01-21 at 6 49 02 PM" src="https://user-images.githubusercontent.com/30560773/213894513-76d6867c-04bd-492c-8671-91e58a495a0f.png">

Modal
<img width="527" alt="Screen Shot 2023-01-15 at 12 37 18 PM" src="https://user-images.githubusercontent.com/30560773/212558076-ce3cf6bb-1e54-41a2-ab50-a58b36ac9ab6.png">

On hover
<img width="522" alt="Screen Shot 2023-01-15 at 12 37 28 PM" src="https://user-images.githubusercontent.com/30560773/212558119-8d43869d-7bff-4b94-8515-1073a109dd91.png">

## Verification Steps

1. Using an account on my local environment, I adopted a tree and added a number of various stewardship activities, making sure to include various combinations to ensure that the position of the delete button is good. 
2. From there, I clicked to delete a stewardship activity and ensured that the Modal popped up with all correct information. 
3. I verified that upon selecting "Delete" in the Modal, the Stewardship activity would no longer appear in the list of selected activities. This was also tested for activities at different positions in the list to ensure that no indexing errors were occurring and the correct activity id was passed.
4. I then created an alternative non-admin account using a different email on my local environment and went to the same previously visited tree to ensure that users and non-admin who did not adopt that tree cannot see the delete button. 
5. I then also added a tree and stewardship activities to the secondary account to make sure that my primary admin account can see the delete button.
